feat(publisher-ers): support ERS v2.x

sebbi08 opened the pull request:
<!--
Thanks for filing a pull request!
Please check off all of the steps as they are completed by replacing [ ] with [x].
-->

- [x] I have read the [contribution documentation](https://github.com/electron/forge/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
- [?] The changes are appropriately documented (if applicable). **_Need help here_**
- [x] The changes have sufficient test coverage (if applicable).
- [x] The testsuite passes successfully on my local machine (if applicable).

With the latest version of the ERS the call to the version api need to have a different structure.

BREAKING CHANGE: Not compatible with ERS 1.x

sebbi08 commented:
Ok, I just retested it, and it is still compatible with version 1.x. So we can remove the "BREAKING CHANGE"

erickzhao reviewed (commented):
Does ERS have a comprehensive list of changes for the v2 line? I'm  curious because the commit messages don't seem to mention this `flavor` change specifically.

https://github.com/ArekSredzki/electron-release-server/compare/v1.5.2...v2.0.0

sebbi08 commented:
Hi no sadly no documentation, I had to look at the responses with postman.

But I had to ditch the version 2.0.0 back to version 1.5.2, but I noticed that the structure of the request is there the new one already.

also looking at the changes from 1.5.0 to the current i could not find any meaningfull change in the version controller
https://github.com/ArekSredzki/electron-release-server/compare/v1.5.0...v2.1.2#diff-4ae54b06789ed67b0ae760dd0b9098cedf855e6ca9d8ac6abe4b59526cc2e07dR46

also i the model there was no such change
https://github.com/ArekSredzki/electron-release-server/compare/v1.5.0...v2.1.2#diff-55d3ecb8588f7504efda7e1d01ea592fc16d2624df33ea1421b92d40a7f1e7c8L1

the only change that i see that might be relevant is
https://github.com/ArekSredzki/electron-release-server/compare/v1.5.0...v2.1.2#diff-178b6d7abb8c0b63db25fff320760527d7dc89439d735cc9d4891a20676d3491R137

and they are due to the Sails/Waterline update

sebbi08 commented:
@ArekSredzki maybe you can help us out here

ArekSredzki commented:
Sorry for missing your message @sebbi08! I happen to be working on this now as well here: https://github.com/electron/forge/pull/3149

ArekSredzki commented:
As you mentioned, the API changes were forced by updating SailJS/WaterlineJS.

Could you please elaborate on what you meant with the following:
> But I had to ditch the version 2.0.0 back to version 1.5.2, but I noticed that the structure of the request is there the new one already.

sebbi08 commented:
@ArekSredzki sorry not the best wording 🙈 

I had updated to 2.x but because of some problem with the encryption I changed it back to version 1.5.2.

But even at that version, the API was already in the new structure.
At least the result of the API calls had the new structure, and it also accepted the new req body structure.  


Closing this in favor of https://github.com/electron/forge/pull/3149
